samcli/lib/sync/flows/function_sync_flow.py

```diff
@@ -125,3 +113,24 @@ class FunctionUpdateStatus(Enum):
     SUCCESS = "Successful"
     FAILED = "Failed"
     IN_PROGRESS = "InProgress"
+
+
+def wait_for_function_update_complete(lambda_client: BaseClient, physical_id: str) -> None:
+    """
+    Parameters
+        ----------
+        lambda_client : boto.core.BaseClient
+            Lambda client that performs get_function API call.
+        physical_id : str
+            Physical identifier of the function resource
+    """
+
+    status = FunctionUpdateStatus.IN_PROGRESS.value
+    while status == FunctionUpdateStatus.IN_PROGRESS.value:
+        response = lambda_client.get_function(FunctionName=physical_id)  # type: ignore
```

Comment:
Instead of ignoring the type can we set it as dict?

Reply:
The type error here is that base client class does not contain get_function method, however boto's create client returns an instance of base client, that's why I'm ignoring the type here!